File: M6/readability.py
def main():
    text = input("Text: ")
    value = grade(text)
    if value <= 1:
        print("Before grade 1")
    elif value < 16:
        print(f"Grade {value}")
    else:
        print("Grade 16+")


def grade(text):
    Alphabet = []
    Points = ['.','?','!']
    for i in range(26):
        Alphabet.append(chr(65 + i))
        Alphabet.append(chr(97 + i))
    letters = 0
    words = 0
    phrases = 0
    for i in range(len(text)):
        if text[i] in Alphabet:
            letters += 1
        elif text[i] == ' ':
            words += 1
        elif text[i] in Points:
            phrases += 1
    words += 1
    index = 0.0588 * (letters/words) * 100 - 0.296 * (phrases/words) * 100 - 15.8
    index = round(index)
    return index

if __name__ == "__main__":
    main()

File: M2/src/readability.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 1000

int readability(const char*);

int main(){

  char *text;
  int index;

  if(!(text = (char*) malloc(sizeof(char) * SIZE))) exit(1);

  printf("Text: ");
  scanf("%1000[^\n]", text);

  index = readability(text);

  if(index < 1) puts("Before Grade 1");
  else if(index >= 16) puts("Grade 16+");
  else printf("Grade: %d\n", index);

  free(text);

  return 0;
}

int readability(const char *text){
  int letters = 0, words = 0, phrases = 0;
  float result;
  for(int i = 0; *(text + i) != '\0'; i++){
    if((*(text + i) >= 'a' && *(text + i) <= 'z') || (*(text + i) >= 'A' && *(text + i) <= 'Z')) letters++;
    else if((*(text + i) == '!') || (*(text + i) == '?') || (*(text + i) == '.')) phrases++;
    if((*(text + i) == ' ') || (*(text + i + 1) == '\0')) words++;
  }
  result = 0.0588 * (((float)letters/words) * 100) - 0.296 * (((float)phrases/words) * 100) - 15.8;
  return (result - (int)result > 0.5)?(int)result + 1:(int) result;
}
